### ISSUE TITLE
[not an a11y-issue]: Alternate Signer Pressing Enter - Found in 508 Audit on form 21-4142

### ISSUE THREAD

**Issue by tbaker1026** [department-of-veterans-affairs/va.gov-team#63436]
## Point of contact

@jeana-adhoc 

## Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

## Details

The 508 office returned an audit with this issue: Changing the setting of the components automatically causes a change of context without advising the user of this behavior.

## Reproduction steps

On `Step 9 of 12` screen, when  “Alternate signer (a person certified to file a claim forms for the Veteran or non-Veteran) radio button checked 6 of 7” is read by assistive technology and when you press the Enter, the screen automatically goes to the next screen “Step 10 of 12: Preparer personal information heading level 2”

<img width="736" alt="three" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/115116356/b542e056-2986-4b52-8f9e-550ae634ddbf">

## Proposed solution or next steps

VA.gov forms support implicit submission with the enter key. This is working as expected.

However, since the question on step 9, is *not* required, if a user presses enter/return it does go to the next screen (This happens currently on any radio/checkbox page). 

What is not made clear though is that *NOTHING* is submitted through this action. We have taken care of this on the back-end so that if nothing is answered, it is assumed they are 3rd party, and the following questions ask the most amount of information for the form. 

We could improve this experience by using the conditionally revealed component that is built with v3 components. But it has accessibility issues of its own right now. 

Still researching solutions.

## References, articles, or WCAG support

1.
2.
3.

## Type of issue

- [X] Screenreader
- [X] Keyboard
- [X] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by jeana-adhoc** [comment]
Follow-up conversations with the 508 office:

I wrote the 508 office and said:

> The forms on va.gov suport "Implicit submission". If all of the requirements of the fields are met on a page, a user should be able to hit enter/return and activate the "submit" button on the page. On all va.gov forms, the "continue" button is of type="submit" since it's storing data on the back-end as a user navigating through the form. No fields were required on this particular page, so whether or not a user interacted with the radio buttons (or selected *any* of the radio buttons), if the enter key is pressed, the form will navigate to the next page.
>
> [Info from the w3c](https://html.spec.whatwg.org/#implicit-submission)
>
> An example of failing WCAG criteria 3.2.2 says "Developers should not use this method to submit a form, and should instead use a submit button, or **[rely on the form's default behavior of submitting when the user hits enter in a text field](https://www.w3.org/TR/WCAG20-TECHS/F36.html)**." 
>
> In the end, I think this is behaving as expected. Let me know if I might be interpreting all of this incorrectly or if we need to disable implicit submission.

Christopher Walker replied:
 
> I’m conflicted about this one.  It feels like an accessibility defect when I press enter on a radio button group and it automatically submits that form even though there are fields underneath it that have never been reached. The screen reader user wouldn’t know they were skipped.   However, doing some google reading, it seems like some people advocate that implicit submission is an accessibility feature.   I’m going to have to talk to some colleagues to see if I am wrong here.   I’ll get back to you on this one.

> We decided that implicit and explicit submissions is a business decision and will not be considered a defect by the 508 office not matter how simple or complex the form.   I’m still a little worried that some of the more complex forms could be problematic for screen reader users but I think the business owners will have to decide on a case by case basis if they want to use it or not.   So in this case, you can mark it as Not a compliance defect.   However I still think you should think about whether or not using it could cause accessibility isssues.

With that said, I'm closing this ticket.
